packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,196 @@
+    // Extend props on the TextField, by passing them through
+    // Downshift's `getInputProps` method
+    return React.Children.map(this.props.children, child => {
+      if (isTextField(child)) {
+        return React.cloneElement(
+          child,
+          getInputProps({
+            id: this.id
+          })
+        );
+      }
+
+      return child;
+    });
+  }
+
+  render() {
+    const {
+      ariaClearLabel,
+      clearInputText,
+      items,
+      itemToString,
+      label,
+      loading,
+      onChange
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          inputValue,
+          isOpen
+        }) => (
+          <div className="ds-u-clearfix ds-c-autocomplete">
+            {this.renderChildren(getInputProps)}
+
+            {isOpen ? (
+              <div className="ds-u-border--1 ds-u-padding--1 ds-c-autocomplete__list">
+                {label &&
+                  !loading && (
+                    <h5
+                      className="ds-u-margin--0 ds-u-padding--1 ds-u-padding-left--2"
+                      id={this.labelId}
+                    >
+                      {label}
+                    </h5>
+                  )}
+
+                <ul
+                  aria-labelledby={label ? this.labelId : null}
+                  className="ds-c-list--bare"
+                  role="listbox"
+                >
+                  {this.filterItems(
+                    items,
+                    inputValue,
+                    getInputProps,
+                    getItemProps,
+                    highlightedIndex
+                  )}
+                </ul>
+              </div>
+            ) : null}
+
+            <Button
+              aria-label={ariaClearLabel}
+              className="ds-u-float--right ds-u-padding-right--0"
+              onClick={clearSelection}
+              size="small"
+              variation="transparent"
+            >
+              {clearInputText}
+            </Button>
+          </div>
+        )}
+      />
+    );
+  }
+}
+
+Autocomplete.defaultProps = {
+  ariaClearLabel: 'Clear typeahead and search again',
+  clearInputText: 'Clear search'
+};
+
+Autocomplete.propTypes = {
+  /**
+   * Screenreader-specific label for the Clear input link. Intended to provide a longer, more descriptive explanation of the link's behavior.
+   */
+  ariaClearLabel: PropTypes.string,
+  children: PropTypes.node,
+  /**
+   * Clear link text that will appear on the page as part of the rendered component
+   */
+  clearInputText: PropTypes.node,
+  /**
+   * Used to determine the string value for the selected item (which is used to compute the `inputValue`.
+   *
+   * Also see: https://github.com/paypal/downshift#itemtostring
+   */
+  itemToString: PropTypes.func,
```

Comment:
This appears to be a required prop based on the current implementation, though could we make it's default the following:

```jsx
(item) => (item ? item.name : '')
```

Reply:
Done. Made `itemToString` a required function, added your example for default.

Reply:
Removed the `.isRequired` attribute per Slack conversatoin.